From 31f4d9a97bb27910a3353d104c2f25118d81d54f Mon Sep 17 00:00:00 2001
Message-Id: <31f4d9a97bb27910a3353d104c2f25118d81d54f.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Naresh Kumar Kachhi <naresh.kumar.kachhi@intel.com>
Date: Thu, 5 Jun 2014 18:08:09 +0530
Subject: [PATCH 125/312] MUST_REBASE [VPG]: drm/i915: DebugFS for Runtime PM

Added DebugFS interface to know display power gate status and usage
count. Should not be used after device is suspended since these will
lead to assertion that device is accessed while suspended.

v2: Removed changes from i915_power_domain_info since checking power
well status requires hardware access and may not be good interface to
check suspend.
Made i915_runtime_pm_info generic debugfs interface to show currrent
runtime state without hardware access.
Updated JIRA tag from Issue to For.

For: GMIN-1416
Change-Id: Id3316ec023597092890db62a7b9384312b30c75c
Signed-off-by: Naresh Kumar Kachhi <naresh.kumar.kachhi@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 2a72368..92a72ab 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -3065,6 +3065,20 @@ static int i915_display_info(struct seq_file *m, void *unused)
 	return 0;
 }
 
+static int i915_runtime_pm_info(struct seq_file *m, void *unused)
+{
+	struct drm_info_node *node = (struct drm_info_node *) m->private;
+	struct drm_device *drm_dev = node->minor->dev;
+	struct device *dev = drm_dev->dev;
+	struct pci_dev *pdev = to_pci_dev(dev);
+
+	seq_printf(m, "%-25s 0x%x\n", "pci state:", pdev->current_state);
+	seq_printf(m, "%-25s %d\n", "usage count:",
+					atomic_read(&dev->power.usage_count));
+
+	return 0;
+}
+
 struct pipe_crc_info {
 	const char *name;
 	struct drm_device *dev;
@@ -4931,6 +4945,7 @@ static const struct drm_info_list i915_debugfs_list[] = {
 	{"i915_energy_uJ", i915_energy_uJ, 0},
 	{"i915_pc8_status", i915_pc8_status, 0},
 	{"i915_power_domain_info", i915_power_domain_info, 0},
+	{"i915_runtime_pm_info", i915_runtime_pm_info, 0},
 	{"i915_display_info", i915_display_info, 0},
 };
 #define I915_DEBUGFS_ENTRIES ARRAY_SIZE(i915_debugfs_list)
-- 
1.7.9.5

